src/components/Flyout/Flyout.test.tsx

```diff
@@ -0,0 +1,67 @@
+import { describe, it, expect, vi, beforeEach, Mock } from 'vitest';
+import { screen, render } from '@testing-library/react';
+import Flyout from './Flyout';
+import { useSelector, useDispatch } from 'react-redux';
+import type { TypedUseSelectorHook } from 'react-redux';
+import { AppDispatch, RootState } from '../../store';
+
+vi.mock('react-redux', async () => {
+  const actual = await vi.importActual('react-redux');
+  return {
+    ...actual,
+    useSelector: vi.fn(),
+    useDispatch: vi.fn(),
+  };
+});
+
+const mockUseSelector = useSelector as unknown as Mock<
+  TypedUseSelectorHook<RootState>
+>;
+const mockUseDispatch = useDispatch as unknown as Mock<() => AppDispatch>;
+
+describe('Flyout', () => {
+  const mockDispatch = vi.fn();
+
+  beforeEach(() => {
+    vi.clearAllMocks();
+
+    mockUseSelector.mockReturnValue({ selectedPokemons: [] });
+    mockUseDispatch.mockReturnValue(mockDispatch);
+  });
+
+  it('test1', () => {
```

Comment:
please add descriptions for tests in this test file instead of `test1, test2, ...` :)